lib/streamlit/runtime/scriptrunner/script_requests.py

```diff
@@ -148,22 +152,30 @@ def request_rerun(self, new_data: RerunData) -> bool:
     def on_scriptrunner_yield(self) -> ScriptRequest | None:
         """Called by the ScriptRunner when it's at a yield point.
 
-        If we have no request, return None.
+        If we have no request or a RERUN request corresponding to one or more fragments,
+        return None.
 
-        If we have a RERUN request, return the request and set our internal
+        If we have a (full script) RERUN request, return the request and set our internal
         state to CONTINUE.
 
         If we have a STOP request, return the request and remain stopped.
         """
-        if self._state == ScriptRequestType.CONTINUE:
-            # We avoid taking a lock in the common case. If a STOP or RERUN
-            # request is received between the `if` and `return`, it will be
-            # handled at the next `on_scriptrunner_yield`, or when
-            # `on_scriptrunner_ready` is called.
+        if self._state == ScriptRequestType.CONTINUE or (
+            # Reruns corresponding to fragments should *not* cancel the current script
+            # run as doing so will affect elements outside of the fragment.
+            self._state == ScriptRequestType.RERUN
+            and self._rerun_data.fragment_id_queue
```

Comment:
This looks redundant with the other changes to this function. Is this a lockless fast-path check?